pa11y/pa11y-ci
# Update notifier

This PR implements [`update-notifier`](https://www.npmjs.com/package/update-notifier) as described in #21.

By default, the `updateCheckInterval` is set to 1 day.

## How to test

The `update-notifier` package follows this rules:
* First launch: save the current date, no update check.
* Next launches:
  * If an update is present in the config file, notify it.
  * If an update is missing, check the date. If the date is upper than the `updateCheckInterval` option, do the update check, store the result in the config file and NOT notify.

So, in order to test the PR you need to:
* Change `package.json` version temporarily:
```diff
{
  "name": "pa11y-ci",
- "version": "2.1.1",
+ "version": "2.1.0",
  ...
}
```
* Run `node ./bin/pa11y-ci.js http://pa11y.org/`
* Edit `~/.config/configstore/update-notifier-pa11y-ci` and insert a fake `lastUpdateCheck` date:
```diff
{
  "optOut": false,
- "lastUpdateCheck": 1538464360361
+ "lastUpdateCheck": 1000000000000
}
```
* Run `node ./bin/pa11y-ci.js http://pa11y.org/` again.
Now, the config file should looking something similar to:
```diff
{
  "optOut": false,
  "lastUpdateCheck": 1538465240023,
+ "update": {
+   "latest": "2.1.1",
+   "current": "2.1.0",
+   "type": "patch",
+   "name": "pa11y-ci"
+ }
}
```
* Run `node ./bin/pa11y-ci.js http://pa11y.org/` for the last time, and a "wild update message appears":
```sh

   ╭─────────────────────────────────────╮
   │                                     │
   │   Update available 2.1.0 → 2.1.1    │
   │    Run npm i pa11y-ci to update     │
   │                                     │
   ╰─────────────────────────────────────╯
```
